sql/server.proto

```diff
@@ -32,4 +32,5 @@ message Session {
   optional int32 syntax = 2 [(gogoproto.nullable) = false];
   // Open transaction.
   optional proto.Transaction txn = 3;
+  optional bool mutates_system_db = 4 [(gogoproto.nullable) = false, (gogoproto.customname) = "MutatesSystemDB"];
```

Comment:
Added comment.
I'm not crazy about required either and since this is entirely controlled by the sql server, I don't think we need to push the logic into the proto itself.